app/src/main/java/com/randomappsinc/studentpicker/presentation/PresentationManager.kt

```diff
@@ -0,0 +1,100 @@
+package com.randomappsinc.studentpicker.presentation
+
+import android.animation.Animator
+import android.animation.AnimatorSet
+import android.animation.ObjectAnimator
+import android.content.Context
+import android.content.res.AssetFileDescriptor
+import android.media.AudioManager
+import android.media.MediaPlayer
+import android.os.Handler
+import android.view.View
+import android.view.animation.AccelerateInterpolator
+import android.view.animation.DecelerateInterpolator
+import com.randomappsinc.studentpicker.R
+import com.randomappsinc.studentpicker.utils.UIUtils
+
+class PresentationManager(
+        private val listener: Listener,
+        private val view: View) {
+
+    interface Listener {
+        fun speakNames()
+    }
+
+    private var player: MediaPlayer? = null
+    private var handler: Handler? = null
+    private val animateNamesTask = ::animateNames
+    private var defaultAnimationLengthMs: Long = 0
+
+    init {
+        handler = Handler()
+        player = MediaPlayer()
+        defaultAnimationLengthMs = view.context.resources.getInteger(R.integer.default_anim_length).toLong()
+    }
+
+    fun playSoundOrAnimations() {
+        view.clearAnimation()
+        view.alpha = 0.0f
+        if ((view.context.getSystemService(Context.AUDIO_SERVICE) as AudioManager)
+                        .getStreamVolume(AudioManager.STREAM_MUSIC) == 0) {
+            playNamesAnimation(defaultAnimationLengthMs)
+        } else {
+            playSound()
+        }
+    }
+
+    private fun playSound() {
+        try {
+            val fileDescriptor: AssetFileDescriptor =
+                    view.context.assets.openFd(PresentationActivity.DRUMROLL_FILE_NAME)
+            player!!.reset()
+            player!!.setDataSource(
+                    fileDescriptor.fileDescriptor,
+                    fileDescriptor.startOffset,
+                    fileDescriptor.length)
+            player!!.prepare()
+            player!!.start()
+        } catch (ex: Exception) {
+            UIUtils.showLongToast(R.string.drumroll_error, view.context)
+        }
+        playNamesAnimation(2600)
```

Comment:
nit: Let's move 2600 to some sort of constant like ANIMATION_DELAY_MS. Maybe have a comment saying that 2.6s is the amount of time you need to wait for the drumroll crash to happen (we want to sync the animating in of the names with that crash).